qutebrowser/commands/runners.py

```diff
@@ -57,11 +57,19 @@ def replace_variables(win_id, arglist):
                                                     QUrl.RemovePassword)
     if '{url:pretty}' in arglist:
         pretty_url = _current_url(tabbed_browser).toString(QUrl.RemovePassword)
+    if '{clipboard}' in arglist:
+        clipboard = utils.get_clipboard()
+    if '{primary}' in arglist:
+        primary = utils.get_clipboard(selection=True)
     for arg in arglist:
         if arg == '{url}':
             args.append(url)
         elif arg == '{url:pretty}':
             args.append(pretty_url)
+        elif arg == '{clipboard}':
+            args.append(clipboard)
+        elif arg == '{primary}':
+            args.append(primary)
```

Comment:
What's a good way of doing that without getting the clipboard contents if they're not needed and without getting the clipboard contents multiple times if the variable is used multiple times?